src/twisted/protocols/test/test_tls.py

```diff
@@ -977,6 +981,54 @@ def disconnected(ign):
         return disconnectDeferred
 
 
+    def test_noCircularReferences(self):
+        """
+        TLSMemoryBIOProtocol doesn't leave circular references that make
+        it alive after connection is closed.
+        """
+        def nObjectsOfType(type):
+            return sum(1 for x in gc.get_objects() if isinstance(x, type))
+
+        gc.disable()
+
+        try:
+            class DummyServerProtocol(Protocol):
```

Comment:
Why is  this protocol "dummy" ? What means dummy?

This looks like a fully working protocol which is closing the connecting first time it receives data.

I have used these types of protocol in production code :)

Please consider adding some docstring to document this code.